### PR TITLE
mvn: add page

### PR DESCRIPTION
**Pull request by jdvr**
----

- [X] The page (if new), does not already exist in the repo.

- [X] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [X] The page has 8 or fewer examples.

- [X] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited

- [X] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: pages/common/mvn.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# mvn
+
+> Runnign Apache Maven.
```
Comment (agnivade) [inline pages/common/mvn.md]: Please improve this description to actually say what does apache maven do.

File: pages/common/mvn.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+- Most common usage is to invoke a life cicle phase:
```
Comment (agnivade) [inline pages/common/mvn.md]: We do not use multiple commands for a single example. If this is the most common use case, I would recommend to split this into 3 separate examples. And probably you can remove the other examples which are not so commonly used.

Comment (jdvr) [reply pages/common/mvn.md]: I have remove the two extra lines, it worths less that the next examples.

File: pages/common/mvn.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# mvn
+
+> Apache Maven. A build automation tool used for JVM environment projects.
```
Comment (agnivade) [inline pages/common/mvn.md]: Please move the 2nd sentence to a new line.

File: pages/common/mvn.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+- Most common usage is to invoke a life cicle phase:
```
Comment (agnivade) [inline pages/common/mvn.md]: typo at cycle.

This description can be improved. No need to mention "Most common usage". It does not say anything about what `mvn package` does. Can you expand a bit on what is meant by a life cycle phase ?

File: pages/common/mvn.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+`mvn package`
+
+- Invoke more that one phase with arguments:
+
+`mvn clean -P a_profile package clean`
```
Comment (agnivade) [inline pages/common/mvn.md]: The phases should be in tokens.

File: pages/common/mvn.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+- Ouput all debug information:
+
+`mvn -X clean -P a_profile package clean`
+
+- Use an alternative pom o directory:
```
Comment (agnivade) [inline pages/common/mvn.md]: o directory ? And please expand pom in brackets.

File: pages/common/mvn.md (hunk 5 above)
Comment (jdvr) [reply pages/common/mvn.md]: Should the "a_profile" be in token?

File: pages/common/mvn.md (hunk 4 above)
Comment (jdvr) [reply pages/common/mvn.md]: It is a little bit difficult to add a phase definition without the maven context, what do you think about this:
"Invoke a Lifecycle phase (a set of steps for the whole project Lifecycle)"?

File: pages/common/mvn.md (hunk 5 above)
Comment (agnivade) [reply pages/common/mvn.md]: Yes.

File: pages/common/mvn.md (hunk 4 above)
Comment (agnivade) [reply pages/common/mvn.md]: we can live with it.

**Comment by hohonuuli** [comment]
Common usages I would recommend to be added:
----
- Delete the target folder, which contains all build outputs:

`mvn clean`

- Compile and build release package, skipping unit tests:

`mvn package -Dmaven.test.skip=true`

- Install the built package in local maven repository:

`mvn install`

File: pages/common/mvn.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+`mvn install`
+`mvn deploy`
+
+- Invoke more that one phase with arguments:
+
+`mvn clean -P a_profile package clean`
+
+- Running an spring boot project with remote debug:
+
+`mvn spring-boot:run -Drun.jvmArguments="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=5005"`
```
Comment (hohonuuli) [inline pages/common/mvn.md]: `spring-boot:run` is pretty esoteric. A more common maven usage would be to run a main class in your project:

----
- Run a class with a main method:

`mvn exec:java -Dexec.mainClass="{{com.example.Main}}" -Dexec.args="{{arg1 arg2}}"`

File: pages/common/mvn.md (hunk 3 above)
Comment (hohonuuli) [reply pages/common/mvn.md]: The maven docs describe mvn as:

"Tool for building and managing Java-based projects."

**Comment by agnivade** [comment]
I am really confused about the `mvn package` command. 

@hohonuuli - your description says `Compile and build release package`
@jdvr - your description says `Most common usage is to invoke a life cycle phase`

Both seem very different to me. I am not a mvn user so I do not know much. However from a layman's standpoint, I would say @hohonuuli's description sounds much more clear to me.

Combining both PRs, I would recommend to have examples in this order - 

`mvn compile`

`mvn -f {{path/to/custom_pom.xml}} compile`

`mvn package`

`mvn install`

`mvn clean`

`mvn exec:java -Dexec.mainClass="{{com.example.Main}}" -Dexec.args="{{arg1 arg2}}"`

**Comment by jdvr** [comment]
This example are missing the fact that pass an argument to a phase like profile on package:
`mvn  -P  {{a_profile}} package`

**Comment by agnivade** [comment]
That's right. Please understand that our scope in tldr is really limited and we have to be judicious in the space we use. We have a maximum limit of 8 examples and we usually try to keep it at 6. So we have to think of the most useful and simplest form of the command invocation first and then move on to the complex ones.

Keeping that in mind, I did not mention passing an argument to the `mvn package` command. However, if you think that it is a really important option that will be needed by beginners, you can add it.

**Comment by jdvr** [comment]
I have tried to join the two PR with the best information,  I've tried to add a generic examples of the commands that new members of my teams used to need.

**Comment by agnivade** [comment]
Hey @jdvr - We always introduce the simplest invocation of the commands first and then move on to more complex ones. You have used `mvn compile` with an argument without mentioning `mvn compile` first. Same with `mvn clean`. The general convention is to mention `mvn compile` and `mvn clean` and explain what it does before moving on to more complex examples. It seems your requirements are a bit specific than the general command invocations.

Also the language of the command descriptions can be improved slightly. If you don't mind, may I edit the PR ?

**Comment by hohonuuli** [comment]
@jdvr Just a note that I've been using Maven for about 20 years, but I've never once used: `mvn -f {{path/to/custom_pom.xml}} compile`. Is that something that you find yourself using day to day? I do however, commonly skip unit tests when building a final package as they can often take a long time to run. perhaps trim the examples to:

`mvn compile`

`mvn package`

`mvn package -Dmaven.test.skip=true`

`mvn install`

`mvn clean`

`mvn exec:java -Dexec.mainClass="{{com.example.Main}}" -Dexec.args="{{arg1 arg2}}"`

**Comment by agnivade** [comment]
Thanks @hohonuuli - I agree with you. I have asked around a couple of people who use maven. And collating all of their feedback, I have made the final changes. It matches almost perfectly with your suggestions. Also, I have added a few more commands which other people have found useful. I had to stretch the page to 8, but I think given the regular usage of the command, it's worth it.

**Comment by jdvr** [comment]
Yes, it looks great. Sorry for including that options, maybe my scope it is to limited :).

**Comment by agnivade** [comment]
Over to you @sbrl

**Pull request merged**